### DISCUSSION TITLE
Reolink camera stream image is not static

### DISCUSSION THREAD

**Discussion by Kate222** [blakeblackshear/frigate#1062] [Ask A Question] [answered]
I am using two cameras with frigate nvr. While Dahua camera works very well with frigate nvr, Reolink camera does not. I have changed Reolink camera to stream with  rtmp and also  used same ffmpeg as what is from frigate document. The rtmp link gives me good result from VLC. However, the reolink camera stream image is not static in frigate nvr. It continues to revolve. Does anyone have good result for reolink camera? It is RLC-410W. Thank you

**Comment by RealKanashii** [comment]
2 reolink cameras here: C2-Pro, RLC-511W

```
ffmpeg:
  # Optional: global ffmpeg args (default: shown below)
  global_args: -hide_banner -loglevel fatal
  # Optional: global hwaccel args (default: shown below)
  # NOTE: See hardware acceleration docs for your specific device
  hwaccel_args: []
  # Optional: global input args (default: shown below)
  input_args: -avoid_negative_ts make_zero -fflags +genpts+discardcorrupt -rtsp_transport tcp -stimeout 5000000 -use_wallclo
ck_as_timestamps 1
  # Optional: global output args
  output_args:
    # Optional: output args for detect streams (default: shown below)
    detect: -f rawvideo -pix_fmt yuv420p
    # Optional: output args for record streams (default: shown below)
    record: -f segment -segment_time 60 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an
    # Optional: output args for clips streams (default: shown below)
    clips: -f segment -segment_time 10 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an
    # Optional: output args for rtmp streams (default: shown below)
    rtmp: -c copy -f flv
    
    ...
    ...
    ...
    
cameras:
  # Name of your camera
  entrada:
    ffmpeg:
      inputs:
        - path: rtsp://xxxxxx:xxxxxxxxxx@xxxxxxxxxxxxx:554/h264Preview_01_sub
          roles:
            - detect
            - clips
            - record
    width: 640
    height: 480
    fps: 5
  # Name of your camera
  terrassa:
    ffmpeg:
      inputs:
        - path: rtsp://xxxxxx:xxxxxxx@xxxxxxxxxxxxx:554/h264Preview_01_sub
          roles:
            - detect
            - clips
            - record
    width: 640
    height: 480
    fps: 5
```

**Reply by RealKanashii** [reply]
be sure to match w/h and fps to avoid green blobs. And 5 fps is enough for detection. This config is for version 0.8.4.

**Reply by Kate222** [reply]
rtsp stream does not work for me. I use home assistant. I have also added reolink camera to configuration.yaml file so I can add it to lovelace card. Reolink camera does not show with rtsp but is fine with rtmp.  Thank you

**Comment by MattEng0** [comment, marked as answer]
Another Reolink user here too and I've managed to get mine to work with the RTMP stream (much more stable that the RTSP stream).

First thing make sure you are using the sub stream and not the main one as that could cause issues with the Width and Height!

Could you post your Frigate config file (making sure to edit out your sensitive details) to see if there is anything in your config file that needs tweaked?

**Reply by Kate222** [reply]
This is my config file for reolink camera. I am not familiar with ffmpeg so it is just a copy and past from frigate file. It was a green screen without it. 
 driveway:
    ffmpeg:
      input_args:
        - -avoid_negative_ts
        - make_zero
        - -fflags
        - nobuffer
        - -flags
        - low_delay
        - -strict
        - experimental
        - -fflags
        - +genpts+discardcorrupt
        - -rw_timeout
        - '5000000'
        - -use_wallclock_as_timestamps
        - '1'
      inputs:
        - path: rtmp://<camera_ip>/bcs/channel0_sub.bcs?channel=0&stream=0&user=xxxx&password=xxxxx
          roles:
            - detect
            - rtmp
    width: 640
    height: 360
    fps: 4

**Reply by Kate222** [reply]
The code positions  have gone wrong when I load it. The original one is correct i.e. with the correct spaces in front.

**Reply by MattEng0** [reply]
Ok I’ll need you to log into the Reolink Camera itself to check the sub streams width and height. To do this log in using the IP address of your browser then log in using the credentials you have there. When you are logged in click on the settings cog in the top right and select stream. You’ll see the sub stream settings on that screen.

My gut is that your height is wrong above. The default would be 640 x 480!

**Reply by Kate222** [reply]
I have just double checked, it is 640x360.

**Reply by MattEng0** [reply]
Ok when you open the stream on VLC what does the codec information say? 

(Tools -> Media Information -> Codec Details)

**Reply by Kate222** [reply]
![image](https://user-images.githubusercontent.com/54869431/117593815-d7de2900-b190-11eb-8110-8bea17771802.png)

**Reply by Kate222** [reply]
I have changed height to 352 and it has worked. The image is now static. Thank you so much.
